_gsocblogs/2022/blog_PODIO_SoumilBaldota.md

```diff
@@ -3,44 +3,74 @@
 
-Next I began to work on a prototype for the Interface which described how the generated code would look like. The prototype was created for classes with one to one relations, one to many relations and vector members. Then I made unittests for the prototype and added them to github actions for continuous integration. 
+---
+
+  
+**Coding Period**
+
+Following the Community Bonding Period I began to work on a prototype for the Interface which described how the generated code would look like. The prototype was created for classes with one to one relations, one to many relations and vector members. Then I made unittests for the prototype and added them to github actions for continuous integration. 
 The prototype can be found at the below link
 [Prototype](https://github.com/soumilbaldota/PODIO_Julia_Interface_Prototype.git)
-  
-*Coding Period*
+
 After the prototype was created. I added an attribute to store the Julia type of a data member described in C++ to the MemberVariable Class and a function to [get the Julia type](https://github.com/AIDASoft/podio/pull/310/files#diff-c129698a9b29360c0e27c5e4f710981b4f99524ad44c039202d750bcf349c834) in [generator_utils.py](https://github.com/AIDASoft/podio/blob/julia/python/generator_utils.py), this was a necessary step to feed the jinja2 templates with the appropriate Julia type for code generation.
 After Julia type was added I made some [unittests](https://github.com/AIDASoft/podio/pull/310/files#diff-61702c3a214182795b1d726c5dc1679a64a10274b7929e4ad4ceaf8dc87c203d) for the same and added them to [test_MemberParser.py](https://github.com/AIDASoft/podio/blob/julia/python/test_MemberParser.py).
-I made jinja2 [templates](https://github.com/AIDASoft/podio/blob/aae66dce1096b1c61bad3b84a12e1678f1593ce3/python/templates/MutableStruct.jl.jinja2) for Julia code generation 
 
+Following the addition of the attributes necessary for building the includes required to feed new templates that will generate Julia code. There is actual preprocessing required for building these includes required by our [Constructor](https://github.com/AIDASoft/podio/blob/59fe1e740ca0a7dbff1180c1425047ed1ab3e027/python/templates/Constructor.jl.jinja2) and the [MutableStruct](https://github.com/AIDASoft/podio/blob/59fe1e740ca0a7dbff1180c1425047ed1ab3e027/python/templates/MutableStruct.jl.jinja2) templates(covered further in the blog). This preprocessing is done by the [**preprocess_for_julia**](https://github.com/AIDASoft/podio/blob/59fe1e740ca0a7dbff1180c1425047ed1ab3e027/python/podio_class_generator.py#L295-L320) function. This functions fills our includes dictionary passed to the templates.
+
+Mutually Recursive Type Declarations in Julia required the use of Parametric types which required the list of parameters required by each Constructor. This data is processed by the [**get_julia_params**](https://github.com/AIDASoft/podio/blob/59fe1e740ca0a7dbff1180c1425047ed1ab3e027/python/podio_class_generator.py#L286-L293)

```

Comment:
Here you could also add a sentence that this is solved with forward declarations in c++ which is not possible in Julia. You can also link to the Julia issue describing the possible workarounds and maybe even briefly say why we chose the parametric types at the end.

You could also go into a few more details to describe which parts are now actually solved with parametric types (and why e.g. vector members are not).

Reply:
Added it in the latest commit